### PR TITLE
Enhance/5308 - Include a "request access" link to Analytics when a user encounters an "Insufficient Permissions" notice

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #5308 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/ReportError.js [hunk 1]
```diff
@@ -151,22 +160,37 @@ export default function ReportError( { moduleSlug, error } ) {
...
+				</Button>
+			) }
 			{ showRetry ? (
 				<Fragment>
 					<Button onClick={ handleRetry }>
 						{ __( 'Retry', 'google-site-kit' ) }
 					</Button>
 					<span className="googlesitekit-error-retry-text">
-						{ __( 'Retry didn’t work?', 'google-site-kit' ) }{ ' ' }
+						{ __( "Retry didn't work?", 'google-site-kit' ) }{ ' ' }
```
Comment (jimmymadon) [inline assets/js/components/ReportError.js:179]: Is this an unnecessary change - perhaps made by the editor?

File: assets/js/components/ReportError.test.js [hunk 2]
```diff
@@ -432,6 +493,6 @@ describe( 'ReportError', () => {
 		expect(
 			getByRole( 'link', { name: /get help/i } )
 		).toBeInTheDocument();
-		expect( queryByText( /retry didn’t work/i ) ).toBeInTheDocument();
+		expect( queryByText( /retry didn't work/i ) ).toBeInTheDocument();
```
Comment (jimmymadon) [inline assets/js/components/ReportError.test.js:496]: I think we prefer keeping the apostrophes to be that special character. So this should be an unnecessary change.

File: assets/js/components/ReportError.test.js [hunk 3]
```diff
@@ -395,7 +456,7 @@ describe( 'ReportError', () => {
 		expect(
 			getByRole( 'link', { name: /get help/i } )
 		).toBeInTheDocument();
-		expect( queryByText( /retry didn’t work/i ) ).not.toBeInTheDocument();
+		expect( queryByText( /retry didn't work/i ) ).not.toBeInTheDocument();
```
Comment (jimmymadon) [inline assets/js/components/ReportError.test.js:459]: Unnecessary change.

**Review by jimmymadon — CHANGES_REQUESTED** [review]
Thanks @hussain-t - this looks great overall. Just one case of duplication and then this should be good to go.

File: assets/js/modules/analytics/datastore/service.js [hunk 4]
```diff
@@ -107,6 +107,31 @@ export const selectors = {
...
+				select( MODULES_ANALYTICS ).getInternalWebPropertyID();
+			const profileID = select( MODULES_ANALYTICS ).getProfileID();
+
+			if ( ! accountID || ! internalWebPropertyID || ! profileID ) {
+				return undefined;
+			}
+
+			const path = escapeURI`/report/report-home/a${ accountID }w${ internalWebPropertyID }p${ profileID }/`;
+
+			return selectors.getServiceURL( state, { path } );
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics/datastore/service.js]: Good catch, thanks, @jimmymadon. Similarly, the Search Console selector should use the `getServiceURL` selector. I have updated that as well.

File: assets/js/components/ReportError.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/components/ReportError.js]: My VS Code was complaining about it. I have reverted now.

File: assets/js/components/ReportError.test.js (hunk 2 above)
Comment (hussain-t) [reply assets/js/components/ReportError.test.js]: My VS Code was complaining about it. I have reverted now.

File: assets/js/components/ReportError.test.js (hunk 3 above)
Comment (hussain-t) [reply assets/js/components/ReportError.test.js]: Same here.

**Review by jimmymadon — APPROVED** [review]
LGTM - thanks @hussain-t!

**Review by techanvil — APPROVED** [review]
LGTM!
